xoto3/lam/finalize.py

```diff
@@ -67,8 +64,30 @@ def __setup_lambda_finalize_hook():
         return
 
     try:
+        aws_lambda_bootstrap = None
         if sys.version_info[0] == 3 and sys.version_info[1] == 7:
-            __py37_finalize_hook()
+            import bootstrap as _aws_lambda_bootstrap  # pylint: disable=import-outside-toplevel,import-error
+
+            aws_lambda_bootstrap = _aws_lambda_bootstrap
+        else:
+            try:
+                from awslambdaric import __version__  # pylint: disable=import-outside-toplevel
+
+                if __version__.split(".")[0] == "2":
+                    import awslambdaric.bootstrap as _aws_lambda_bootstrap  # pylint: disable=import-outside-toplevel,import-error
+
+                    aws_lambda_bootstrap = _aws_lambda_bootstrap
+                else:
+                    logger.error(
+                        f"Unimplemented for version of AWS Lambda Runtime Interface Client: {__version__}"
+                    )
+            except ImportError:
+                logger.error(
+                    "Failed to import awslambdaric. A new implementation might be needed for this runtime."
+                )
+
+        if aws_lambda_bootstrap is not None:
+            __install_post_invocation_hooks(aws_lambda_bootstrap)
```

Comment:
This is definitely not the prettiest, but it works. I don't foresee this changing any time soon given the frequency of changes in https://github.com/aws/aws-lambda-python-runtime-interface-client/ but we should definitely tidy this up if the need ever arises. Chances are we will be able to drop 3.7 due to it reaching EOL when that day comes.